client.go

```diff
@@ -221,10 +222,14 @@ func (c *Client) Do(req *http.Request, v interface{}) error {
 	contentEncoding := strings.ToLower(res.Header.Get("Content-Encoding"))
 	if contentEncoding == "gzip" {
 		gzipReader, err := gzip.NewReader(bodyReader)
+		// TODO clean this up
```

Comment:
Need to clean this up and document what it's doing. It's a bit confusing as I have not been able to find a way to skip this when the body is empty. The only way I could do it is to apply the gzip reader and get back an `EOF` error.  In that case, we fall through.